tests/queryset_pickle/tests.py

```diff
@@ -115,6 +115,38 @@ def test_pickle_prefetch_related_idempotence(self):
         groups = pickle.loads(pickle.dumps(groups))
         self.assertSequenceEqual(groups, [g])
 
+    def test_pickle_prefetch_queryset_not_evaluated(self):
+        Group.objects.create(name='foo')
+        groups = Group.objects.prefetch_related(
+            models.Prefetch('event_set', queryset=Event.objects.order_by('id'))
+        )
+        list(groups)  # evaluate QuerySet
+        with self.assertNumQueries(0):
+            pickle.loads(pickle.dumps(groups))
+
+    def test_pickle_prefetch_queryset_usable_outside_of_prefetch(self):
+        Group.objects.create(name='foo')
+        events = Event.objects.order_by('id')
+        # Just because we pass 'events' to this Prefetch shouldn't mean it
```

Comment:
Try to avoid "we" in comments. It usually results in more concise phrasing. e.g. "The Prefetch caches its results when pickled if ..."